lib/Basic/LangOptions.cpp

```diff
@@ -157,6 +158,8 @@ std::pair<bool, bool> LangOptions::setTarget(llvm::Triple triple) {
     addPlatformConditionValue(PlatformConditionKind::OS, "FreeBSD");
   else if (triple.isOSWindows())
     addPlatformConditionValue(PlatformConditionKind::OS, "Windows");
+  else if (triple.isWindowsCygwinEnvironment())
+    addPlatformConditionValue(PlatformConditionKind::OS, "Cygwin");
```

Comment:
On Cygwin, `triple.isOSWindows()` and `triple.isWindowsCygwinEnvironment()` are all `true`.
The condition order should be changed. After changed, `os(Windows)` will be `false` and only `os(Cygwin)` will be `true`.